### PR TITLE
Decrease size of upload for progress reporting test to avoid memory issues on devops agents; Fix progress test flakiness

### LINKED ISSUE #9282: [BUG] StageBlockAsync_ProgressReporting live test failure
https://dev.azure.com/azure-sdk/internal/_build/results?buildId=221925&view=ms.vss-test-web.build-test-results-tab&runId=7623588&resultId=100565&paneView=debug

Looking at the attachments tab that shows the request/response logging, the issue appears to be that in the case of a retry, the progress will be reset, which makes the test assertion that each progress update is greater or equal to the previous one fail. I think this behavior of resetting the progress is technically correct, but we might want to consider if that is the best user experience. I think the more common pattern would be holding the progress that we report to be at its current level until it passes it again during the retry. Customers could implement this on their own, but it is unlikely they would run into this during testing.

### PR DESCRIPTION
**Pull request by JoshLove-msft**
We don't actually need to upload 1GB to hit the parallel code path.

Update ProgressReporting tests to remove the monotonic increasing assertions, as retries will reset the streams to 0.

Add retry after to datalake container creation to work around service bug.

Fixes https://github.com/Azure/azure-sdk-for-net/issues/9282

### CODE REVIEW

**Review by seanmcc-msft — APPROVED** [review]

**Review by amnguye — APPROVED** [review]

File: sdk/storage/Azure.Storage.Files.DataLake/tests/DataLakeTestBase.cs [hunk 1]
```diff
@@ -103,7 +103,28 @@ public DataLakeServiceClient GetServiceClient_OAuth()
             }
 
             DataLakeFileSystemClient fileSystem = InstrumentClient(service.GetFileSystemClient(fileSystemName));
-            await fileSystem.CreateAsync(metadata: metadata, publicAccessType: publicAccessType);
+
+            // due to a service issue, if the initial container creation request times out, subsequent requests
+            // can return a ContainerAlreadyExists code even though the container doesn't really exist.
+            // we delay until after the service cache timeout and then attempt to create the container one more time.
+            // If this attempt fails, we let the exception propagate.
+            // TODO Note this issue will be fixed in the 72 rollout. After that time, this try/catch can be removed.
```
Comment (tg-msft) [inline sdk/storage/Azure.Storage.Files.DataLake/tests/DataLakeTestBase.cs]: Nit - file a bug on removing this.

File: sdk/storage/Azure.Storage.Common/tests/Shared/TestConstants.cs [hunk 2]
```diff
@@ -19,6 +19,10 @@ public class TestConstants
         public const string Https = "https";
         public const int HttpPort = 443;
 
+        // service cache timeout is 60s, but adding a buffer to be safe
+        public const int DataLakeRetryDelay = 70000;
```
Comment (tg-msft) [inline sdk/storage/Azure.Storage.Common/tests/Shared/TestConstants.cs]: Nit - rename to DataLakeCacheInvalidationRetryDelay or similar to make it clear that other folks shouldn't use this particular constant.

File: sdk/storage/Azure.Storage.Common/tests/Shared/StorageTestBase.cs [hunk 3]
```diff
@@ -423,5 +438,33 @@ internal void AssertResponseHeaders(TestConstants constants, SasQueryParameters
...
         }
+
+        protected async Task<T> RetryAsync<T>(
+            Func<Task<T>> operation,
+            Func<RequestFailedException, bool> shouldRetry,
+            int retryDelay = TestConstants.RetryDelay,
+            int retryAttempts = Constants.MaxReliabilityRetries) =>
+            await RetryAsync(Mode, operation, shouldRetry, retryDelay, retryAttempts);
+
+        public static async Task<T> RetryAsync<T>(
```
Comment (tg-msft) [inline sdk/storage/Azure.Storage.Common/tests/Shared/StorageTestBase.cs]: Why do we need the `static` versions?

File: sdk/storage/Azure.Storage.Blobs/tests/BlobClientTests.cs [hunk 4]
```diff
@@ -757,20 +757,16 @@ public async Task UploadAsync_ProgressReporting()
...
             // Act
-            await blob.UploadAsync(content: stream, progressHandler: progress);
+            await blob.UploadAsync(content: stream, progressHandler: progress, transferOptions: options);
 
             // Assert
             Assert.IsFalse(progress.List.Count == 0);
 
-            for ( int i = 1; i < progress.List.Count; i++)
-            {
-                Assert.IsTrue(progress.List[i] >= progress.List[i - 1]);
```
Comment (tg-msft) [inline sdk/storage/Azure.Storage.Blobs/tests/BlobClientTests.cs]: So we're no longer going to check these are ordered?

**Pull request merged**